novawallet/Modules/Vote/Governance/ReferendumVoteSetup/ReferendumVoteSetupPresenter.swift

```diff
@@ -443,6 +443,13 @@ extension ReferendumVoteSetupPresenter: ReferendumVoteSetupPresenterProtocol {
 }
 
 extension ReferendumVoteSetupPresenter: ReferendumVoteSetupInteractorOutputProtocol {
+    func didReceiveVotingReferendumsState(_ state: ReferendumsState) {
+        votesResult = state.voting
+        referendum = state.referendums[referendumIndex]
+
+        refreshLockDiff()
```

Comment:
I would check whether there are changes and after that refresh the locks. Otherwise we endup refreshing each update of the referendum list